### PR TITLE
Add Media object Align Start modifier

### LINKED ISSUE #1792: Add Media object Align Start modifier
The use case is the "Books we've written" content. In my opinion, the horizontal misalignment of the book title heading feels a bit off in the "current solution" as shown below. To me, the layout feels better when the book title headings align horizontally as shown in the "proposed solution" below.

Current solution | Proposed solution
--- | ---
Screen Shot 2022-05-12 at 5 16 30 AM | Screen Shot 2022-05-12 at 5 16 50 AM

### PR DESCRIPTION
**Pull request by gerardo-rodriguez**
## Overview

This PR proposes adding an `o-media--align-start` modifier. See #1792 for an example use case.

## Screenshots

Screen Shot 2022-05-12 at 5 34 33 AM

## Testing

1. Review https://deploy-preview-1791--cloudfour-patterns.netlify.app/?path=/docs/objects-media--align-start#align-start

---

- Closes #1792

### CODE REVIEW

**Review by Paul-Hebert — DISMISSED** [review]
Code looks good. Part of me wonders if this should be a utility class but it doesn't seem like a big deal either way. 

The proposed design change in the Wordpress site makes sense to me. I'll defer to Tyler on that design change, but this seems like a reasonable feature to add to the pattern library either way.

**Comment by gerardo-rodriguez** [comment]
> Part of me wonders if this should be a utility class

Either works for me. 🤷

File: src/objects/media/media.stories.mdx
Comment (tylersticka) [inline src/objects/media/media.stories.mdx:184]: Some suggestions:

- Change the heading to be more generic in case we want to add more alignment modifiers in the future. This is also consistent with the Deck object heading with a similar purpose.
- Explain how this differs from the usual alignment.
- Link to the complimentary Deck object modifier for additional context.
- Cap line lengths consistently to 80 characters. For MDX files I don't think breaking lines manually is a requirement at all, but when we do, adhering to [the long-term standard](https://softwareengineering.stackexchange.com/a/148678) is probably best.

```suggestion
## Alignment

By default, the Media object optimizes its alignment based on its content alone.
This may result in inconsistent alignment when multiple Media objects with
inconsistent content lengths are displayed side by side.

You may add the `o-media--align-start` modifier to force consistent alignment
between adjacent Media objects. This pairs nicely with [the Deck object modifier
of the same name](/docs/objects-deck--block-alignment).
```

File: src/objects/media/media.stories.mdx
Comment (tylersticka) [inline src/objects/media/media.stories.mdx:214]: Should we add a "Template Properties" section now that we have several?

**Review by tylersticka — COMMENTED** [review]
Modifier looks good, some documentation suggestions inline.

File: src/objects/media/media.stories.mdx
Comment (gerardo-rodriguez) [reply src/objects/media/media.stories.mdx:254]: @tylersticka Good idea!

It [this](https://github.com/cloudfour/cloudfour.com-patterns/pull/1791/commits/3d8522e5f0b8c4956a12ebc10cce206cbb21c817) what you had in mind? It's still WIP (I'll have to finish tomorrow), but I am finding it a bit odd I need to connect the `<ArgsTable>` to a specific story. Should I add an `<ArgsTable>` to all nine individual stories? 🤔 

Screen Shot 2022-05-12 at 2 28 08 PM

Comment (tylersticka) [reply src/objects/media/media.stories.mdx:254]: @gerardo-rodriguez `ArgsTable` is a good choice if the story's properties are the same as the template's properties. In this case, I'm skeptical that `imgSrc` and `text` are template properties?

If in doubt, a simple list will suffice. Here's an example: https://v-next--cloudfour-patterns.netlify.app/?path=/docs/components-alert--basic#template-properties

Comment (gerardo-rodriguez) [reply src/objects/media/media.stories.mdx:254]: [Added](https://github.com/cloudfour/cloudfour.com-patterns/pull/1791/commits/0a9073b95fd889f127def9245d2252bdb547b5b7)!

Screen Shot 2022-05-13 at 5 34 24 AM

**Comment by gerardo-rodriguez** [comment]
@Paul-Hebert @tylersticka I've updated the PR, ready for another review, thanks! 🙂

**Review by tylersticka — COMMENTED** [review]
Will approve once CI error is resolved (leftover `ArgsTable` import).

**Comment by gerardo-rodriguez** [comment]
> Will approve once CI error is resolved (leftover `ArgsTable` import).

@tylersticka I think your page is cached. I pushed up [the commit to remove the `ArgsTable` import](https://github.com/cloudfour/cloudfour.com-patterns/pull/1791/commits/fdb008740bd28d29a6cd2f26f302ec5a310b7a85) yesterday. 🤔

**Review by tylersticka — APPROVED** [review]

**Pull request merged**
